File: app/src/main/java/closer/vlllage/com/closer/handler/group/PhoneNumberHandler.java
package closer.vlllage.com.closer.handler.group;

import com.google.i18n.phonenumbers.NumberParseException;
import com.google.i18n.phonenumbers.PhoneNumberUtil;
import com.google.i18n.phonenumbers.Phonenumber;

import java.util.Locale;

import closer.vlllage.com.closer.pool.PoolMember;

public class PhoneNumberHandler extends PoolMember {

    private static final PhoneNumberUtil phoneNumberUtil = PhoneNumberUtil.getInstance();

    public boolean isValidNumber(String phoneNumber) {
        try {
            return phoneNumberUtil.isValidNumber(phoneNumberUtil.parse(phoneNumber, Locale.US.getCountry()));
        } catch (NumberParseException e) {
            e.printStackTrace();
            return false;
        }
    }

    public String normalize(String phoneNumber) {
        try {
            Phonenumber.PhoneNumber number = phoneNumberUtil.parse(phoneNumber, Locale.US.getCountry());
            return phoneNumberUtil.format(number, PhoneNumberUtil.PhoneNumberFormat.INTERNATIONAL);
        } catch (NumberParseException e) {
            e.printStackTrace();
        }

        return null;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/group/PhoneNumberHandler.kt
package closer.vlllage.com.closer.handler.group

import com.queatz.on.On
import com.google.i18n.phonenumbers.NumberParseException
import com.google.i18n.phonenumbers.PhoneNumberUtil
import java.util.*

class PhoneNumberHandler constructor(private val on: On) {

    fun isValidNumber(phoneNumber: String): Boolean {
        return try {
            phoneNumberUtil.isValidNumber(phoneNumberUtil.parse(phoneNumber, Locale.US.country))
        } catch (e: NumberParseException) {
            e.printStackTrace()
            false
        }

    }

    fun normalize(phoneNumber: String): String? {
        return try {
            val number = phoneNumberUtil.parse(phoneNumber, Locale.US.country)
            phoneNumberUtil.format(number, PhoneNumberUtil.PhoneNumberFormat.INTERNATIONAL)
        } catch (e: NumberParseException) {
            e.printStackTrace()
            null
        }
    }

    companion object {
        private val phoneNumberUtil = PhoneNumberUtil.getInstance()
    }
}
